receiver/azuremonitorreceiver/README.md

```diff
@@ -23,7 +23,9 @@ The following settings are required:
 
 The following settings are optional:
 
-- `auth` (default = service_principal): Specifies the used authentication method. Supported values are `service_principal`, `workload_identity`, `managed_identity`, `default_credentials`.
+- `token_provider` or `auth` (default = service_principal): Specifies the used authentication method:
```

Comment:
I think it can be clearer if you keep token_provider and auth separated bullet points. You can have a little duplication here no problem as I think I will remove auth later on.
Can you also put a deprecated flag on the auth one?

Reply:
Yes, I have added it in the README, and also added a warning in the code:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/356c20ca906cf6fc35f713334a5d32d23ae35264/receiver/azuremonitorreceiver/scraper.go#L189